### DISCUSSION TITLE
setRequestLocale Clarification

### DISCUSSION THREAD

**Discussion by jd-carroll** [amannn/next-intl#1629] [General]
The [documentation](https://next-intl.dev/docs/getting-started/app-router/with-i18n-routing#add-setrequestlocale-to-all-relevant-layouts-and-pages) states that we should add `setRequestLocale` to **"all relevant layouts and pages"**.

How are all "relevant" layouts and pages defined?

If my app dir structure follows the `/[local]/*` format, does this mean that `setRequestLocale` needs to be added to every `layout.tsx` **AND** `page.tsx` **AND** `loading.tsx` **AND** `not-found.tsx` **AND** `error.tsx` **AND** `template.tsx` **AND** `default.tsx` under the `/[locale]` folder? (should `route.tsx` also be included in this list?)

\- OR -

Is it possible to add `setRequestLocale` to the `/[locale]/layout.tsx` **only** and not need to add it anywhere else? <br/>(I'm guessing not, but wanted to check)

**Comment by amannn** [comment]
Unfortunately it currently really requires being added to every entry point that receives `params`, not just a wrapping layout.

The good news is that we're very close to a better solution that will make all of this obsolete and support static rendering without any workarounds like `setRequestLocale`: https://github.com/amannn/next-intl/issues/663#issuecomment-2478719013

I hope to have more to share on this soon!

**Comment by jd-carroll** [comment]
Awesome, thank you! Kind of what I figured
